x/participationrewards/types/protocol_data.go

```diff
@@ -131,6 +131,54 @@ func UnmarshalProtocolData(datatype ProtocolDataType, data json.RawMessage) (Pro
 		}
 
 		return &upd, nil
+	case ProtocolDataTypeCrescentParams:
+		cpd := CrescentParamsProtocolData{}
+		err := json.Unmarshal(data, &cpd)
+		if err != nil {
+			return nil, fmt.Errorf("unable to unmarshal intermediary CrescentParamsProtocolData: %w", err)
+		}
+		var blank CrescentParamsProtocolData
+		if reflect.DeepEqual(cpd, blank) {
+			return nil, fmt.Errorf("unable to unmarshal CrescentParamsProtocolData from empty JSON object")
+		}
+
+		return &cpd, nil
+	case ProtocolDataTypeCrescentPool:
+		cpd := CrescentPoolProtocolData{}
+		err := json.Unmarshal(data, &cpd)
+		if err != nil {
+			return nil, fmt.Errorf("unable to unmarshal intermediary CrescentPoolProtocolData: %w", err)
+		}
+		var blank CrescentPoolProtocolData
+		if reflect.DeepEqual(cpd, blank) {
+			return nil, fmt.Errorf("unable to unmarshal CrescentPoolProtocolData from empty JSON object")
+		}
+
+		return &cpd, nil
+	case ProtocolDataTypeCrescentPoolCoinSupply:
+		cpd := CrescentPoolCoinSupplyProtocolData{}
+		err := json.Unmarshal(data, &cpd)
+		if err != nil {
+			return nil, fmt.Errorf("unable to unmarshal intermediary CrescentPoolCoinSupplyProtocolData: %w", err)
+		}
+		var blank CrescentPoolCoinSupplyProtocolData
+		if reflect.DeepEqual(cpd, blank) {
+			return nil, fmt.Errorf("unable to unmarshal CrescentPoolCoinSupplyProtocolData from empty JSON object")
+		}
+
+		return &cpd, nil
+	case ProtocolDataTypeCrescentReserveAddressBalance:
+		cpd := CrescentReserveAddressBalanceProtocolData{}
+		err := json.Unmarshal(data, &cpd)
+		if err != nil {
+			return nil, fmt.Errorf("unable to unmarshal intermediary CrescentReserveAddressBalanceProtocolData: %w", err)
+		}
+		var blank CrescentReserveAddressBalanceProtocolData
+		if reflect.DeepEqual(cpd, blank) {
+			return nil, fmt.Errorf("unable to unmarshal CrescentReserveAddressBalanceProtocolData from empty JSON object")
+		}
+
```

Comment:
Is there any way to do this in a more scalable way? Every new PD is another 10 duplicated lines.